html/user/host_stats.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2018 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+// show statistics of the host population
+
+require_once("../inc/util.inc");
+require_once("../inc/cache.inc");
+
+$min_credit = .1;   // only count hosts with this much RAC
+$total_eac = 0;
+
+function show_type($s, $x) {
+    global $total_eac;
+    $p = number_format(100*$x->c/$total_eac, 4);
```

Comment:
On a new project (admittedly a edge case), this causes a division by zero error and the following is output in the PHP.  

> Warning: Division by zero in /root/project/html/user/host_stats.php on line 30
> 
> Warning: Division by zero in /root/project/html/user/host_stats.php on line 30
> 
> Warning: Division by zero in /root/project/html/user/host_stats.php on line 30

Something like the following will prevent this:
```
function show_type($s, $x) {
    global $total_eac;
    if ( $total_eac == 0 ) {
        $p = 0;
    } else {
        $p = number_format(100*$x->c/$total_eac, 4);
    }
    row_array(array($s, $x->n, "$p %"));
}
```